lib/netext/httpext/transport.go

```diff
@@ -115,9 +122,28 @@ func (t *transport) measureAndEmitMetrics(unfReq *unfinishedRequest) *finishedRe
 			tags["status"] = "0"
 		}
 	} else {
+		cleanURL := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}.Clean()
 		if enabledTags.Has(stats.TagURL) {
-			u := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}
-			tags["url"] = u.Clean()
+			// Tag the first request with the original clean URL and
+			// subsequent ones (e.g. part of a redirect chain or digest auth)
+			// with the updated clean URL.
+			if t.origURL != nil && t.numProcReqs == 1 {
+				tags["url"] = t.origURL.Clean()
+			} else {
+				tags["url"] = cleanURL
```

Comment:
hmm I might be missing something, but if `t.numProcReqs == 1`, in what would cause `t.origURL.Clean()` be different than `cleanURL`?

Reply:
ah, nevermind, I should read the PR description before making stupid comments... :man_facepalming: :sweat_smile:

Reply:
Yeah, sorry, maybe I wasn't clear enough. The thing is that `cleanURL` is based on the current request being processed as received in `RoundTrip()`, which doesn't have the `username:password`, so cleaning it would cause the original URL as it appears in the script to be tagged without the masked credentials (e.g. `https://<host><path>`), which is inconsistent with the current behavior.

Reply:
sorry, somehow missed you replied here before posting https://github.com/loadimpact/k6/pull/1570#issuecomment-662941482